core/src/main/java/org/infinispan/util/concurrent/locks/impl/PerKeyLockContainer.java

```diff
@@ -64,10 +67,20 @@ public InfinispanLock getLock(Object key) {
    public void release(Object key, Object lockOwner) {
       lockMap.computeIfPresent(key, (ignoredKey, lock) -> {
          lock.release(lockOwner);
-         return !lock.isLocked() ? null : lock; //remove it if empty
+         return lock.isLocked() ? lock : null; //remove it if empty
       });
    }
 
+   @Override
+   public void releaseAll(Object lockOwner) {
+      //bulk operation are dangerous with concurrent hash map
```

Comment:
the later. I tried to use `replaceAll` but it didn't work very well.

Reply:
Could it be because the return value from your lambda was sometimes `null`? `replaceAll` doesn't allow removing values according to the javadoc. In any case, a vague comment about some bulk operations possibly having unspecified problems doesn't add anything IMO.